token/client/src/token.rs

```diff
@@ -765,23 +772,39 @@ where
     }
 
     /// Close account into another
-    pub async fn close_account<S: Signer>(
+    pub async fn close_account<S: Signers>(
         &self,
         account: &Pubkey,
         destination: &Pubkey,
-        authority: &S,
+        authority: &Pubkey,
+        signing_keypairs: &S,
     ) -> TokenResult<T::Output> {
-        self.process_ixs(
-            &[instruction::close_account(
-                &self.program_id,
-                account,
-                destination,
-                &authority.pubkey(),
-                &[],
-            )?],
-            &[authority],
-        )
-        .await
+        let signing_pubkeys = signing_keypairs.pubkeys();
+        let multisig = if signing_pubkeys == &[*authority] {
+            signing_pubkeys.iter().collect::<Vec<_>>()
+        } else {
+            vec![]
+        };
```

Comment:
oki this is somewhat better, and in line with your suggestion to pass through the empty array when not a multisig

Reply:
haha jk this is egregiously wrong:
* i got the branch arms backwards
* `process_command` pushes the config signer onto `bulk_signers` without checking if its there yet, so in many cases the vec is the payer twice
* having multiple signers in the array is necessary *but not sufficient* evidence that this is a multisig transaction. the simplest case is where payer and authority differ

following the flow from `main`, `bulk_signers` gets multisig signers first, then the fee payer, then possibly a nonce authority, then (in the case of `Command::Close`, eg) close authority, which by default is the payer a second time

there are all kind of weird edge cases possible. eg, if we dedupe payers and it reduces to two addresses, one of which is the authority, we cant know whether the non-authority is payer or whether authority is payer and non-authority is a 1/1 multisig. we can check payer on the `Token` object, but then it depends on whether the user initialized it properly

im inclined to pass all signers to the instruction and let god (token) sort them out

Reply:
i thought a bit more about this and decided the fixed version of this is actually fine. its not the token functions responsibility to make sure the keypairs array is carefully assembled, it just needs to sign with whats given, and expecting to see just the authority key, or some multisig/nonce stuff, is reasonable. its the *callers* responsibility to pass a good array

Reply:
one thing id like is if i could make it a helper function but the lifetime shit is too weird for me

Reply:
Yeah I think that's reasonable. The only footgun with this is a weird situation in which a multisig is a signer for itself. As in, I create a multisig at `multsig111111111111....` and inside, one of its signers is `multisig1111111...`. But maybe we just decide to make life hard for someone like that?

Reply:
im not sure what the issue would be even in this case. the flow is "if `signing_keypairs` is one key that matches the authority, the authority param will mark it as a signer of the instruction. if its anything else, everything is marked as a signer. then all signers sign the transaction unconditionally, in addition to whatever payer may be designated on the `Token` object"

Reply:
incidentally im moving the if/else into a function in my next pr instead of copying this logic around everywhere